src/averageAges.js

```diff
@@ -37,7 +42,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const countFemale = people.filter(el => el.sex === 'f');
+
+  const isMom = withChildren ? (
+    countFemale
+      .filter(name => people.some(el => el.mother === name.name))
+  ) : countFemale;
+
+  const sumFemale = isMom
+    .reduce((accum, age) => accum + (age.died - age.born), 0);
```

Comment:
And also you should format here according to the checklist (p.2)
const humans = people.reduce((a, b) => (
  a + b
), 0);